src/test/regress/expected/multi_prepare_plsql.out

```diff
@@ -509,119 +478,570 @@ SELECT * FROM temp_table ORDER BY key, value;
    4 |      
    5 |    50
    5 |      
-(16 rows)
+(22 rows)
 
 -- check router executor select
-CREATE OR REPLACE FUNCTION partition_column_select(key_arg int) RETURNS TABLE(key int, value int) AS $$
+CREATE FUNCTION router_partition_column_select(key_arg int)
+	RETURNS TABLE(key int, value int) AS $$
 DECLARE
 BEGIN
     RETURN QUERY
 	SELECT
-		temp_table.key,
-		temp_table.value
+		plpgsql_table.key,
+		plpgsql_table.value
 	FROM
-		temp_table
+		plpgsql_table
 	WHERE
-		temp_table.key = key_arg
+		plpgsql_table.key = key_arg
 	ORDER BY
 		key,
 		value;
 END;
 $$ LANGUAGE plpgsql;
-SELECT partition_column_select(1);
- partition_column_select 
--------------------------
+SELECT router_partition_column_select(1);
+ router_partition_column_select 
+--------------------------------
  (1,10)
  (1,)
 (2 rows)
 
-SELECT partition_column_select(2);
- partition_column_select 
--------------------------
+SELECT router_partition_column_select(2);
+ router_partition_column_select 
+--------------------------------
  (2,20)
  (2,)
 (2 rows)
 
-SELECT partition_column_select(3);
- partition_column_select 
--------------------------
+SELECT router_partition_column_select(3);
+ router_partition_column_select 
+--------------------------------
  (3,30)
  (3,)
 (2 rows)
 
-SELECT partition_column_select(4);
- partition_column_select 
--------------------------
+SELECT router_partition_column_select(4);
+ router_partition_column_select 
+--------------------------------
  (4,40)
  (4,)
 (2 rows)
 
-SELECT partition_column_select(5);
- partition_column_select 
--------------------------
+SELECT router_partition_column_select(5);
+ router_partition_column_select 
+--------------------------------
  (5,50)
  (5,)
 (2 rows)
 
--- 6th execution is failing. We don't want to run the failing test because of
--- changing output. After implementing this feature, uncomment this.
--- SELECT partition_column_select(6);
+-- FIXME: 6th execution is failing. We don't want to run the failing test
```

Comment:
Probably! We need to reach [plansource->num_custom_plans](https://github.com/postgres/postgres/blob/ee943004466418595363d567f18c053bae407792/src/backend/utils/cache/plancache.c#L1027) (or another variable to replicate counting)  and also evaluate that if we can actually run the given query at 6th execution or not. But, I think we can leave it at outside of this PR's scope.